From d8f21366d2b9bd72faf078099d4ba8146396ac59 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Wed, 7 Dec 2016 13:54:18 +0000
Subject: [PATCH] subversion: Fix issues in LDFLAGS sed manipulation

The existing sed expression can match expressions like
--sysroot=/some/path/xxx-linux/ which clearly isn't intended and injects
incorrect paths into LDFLAGS.

Fix this in the same way we address the problem in CFLAGS.

RP 2016/12/7
Upstream-Status: Pending

Rebase 1.12.0
Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 build/ac-macros/serf.m4 | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build/ac-macros/serf.m4 b/build/ac-macros/serf.m4
index 4637430..126eb74 100644
--- a/build/ac-macros/serf.m4
+++ b/build/ac-macros/serf.m4
@@ -185,7 +185,7 @@ AC_DEFUN(SVN_SERF_PKG_CONFIG,
           SVN_SERF_INCLUDES=[`$PKG_CONFIG $serf_pc_arg --cflags-only-I`]
           SVN_SERF_LIBS=[`$PKG_CONFIG $serf_pc_arg --libs-only-l`]
           dnl don't use --libs-only-L because then we might miss some options
-          LDFLAGS=["$LDFLAGS `$PKG_CONFIG $serf_pc_arg --libs | $SED -e 's/-l[^ ]*//g'`"]
+          LDFLAGS=["$LDFLAGS `$PKG_CONFIG $serf_pc_arg --libs | $SED -e 's/ -l[^ ]*//g' -e 's/^-l[^ ]*//g'`"]
           break
         else
           AC_MSG_RESULT([no])
